### PR TITLE
upgrade eslint/prettier, add `log` override & implement single-file bundling

### PR DESCRIPTION
**Pull request by zackschuster**
(also includes some minor jsdoc enhancements left over from the last PR)

file this one under the "promising first step" kind of PR, where the rollup bundle seems to emit pretty cleanly & passes tests, but needs some real-world usage to verify (which i can provide). as such, the added files currently are `.npmignore`'d, so it won't immediately affect the published module.

given that rollup can work with commonjs modules "pretty well", converting to ESM possibly/probably isn't necessary, at least in the near future. i'd still encourage it at some point, but from my vantage point landing this would be more beneficial, as it provides an easily-referenced file for deployment/testing while setting up the machinery for if we ever decide to convert.

no new errors or code fixes from the eslint/prettier update.

the `log` override is for azure function apps -- `console.log` doesn't work there, forcing you to log everything through the function's `context` object. i've added a line of documentation in the README; unsure where else to document it.

### CODE REVIEW

**Comment by zackschuster** [comment]
tests are failing on node 7 due to the eslint update. considering that [node 7 has been EOL for a year](https://github.com/nodejs/Release#end-of-life-releases), i recommend dropping it from travis. otherwise i can downgrade eslint.

**Comment by zackschuster** [comment]
went ahead and updated `.travis.yml` to test on all the currently supported versions of node, as well as to run tests against the bundle.

for kicks i tried dropping the c++ toolchain (there for old node-gyp reasons, i'm guessing) and everything appears to be running fine. also cuts a huge chunk of time off the builds, hooray!

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-  apt:
-    sources:
-      - ubuntu-toolchain-r-test
-    packages:
-      - g++-4.8
+  - "6"
+  - "8"
+  - "10"
+script:
+  - yarn test
```
Comment (eleith) [inline .travis.yml]: is it possible to use npm instead?

File: .npmignore [hunk 2]
```diff
@@ -4,3 +4,7 @@
 /.npmignore
 /.travis.yml
 /npm-debug.log
+/email.bundle.js
```
Comment (eleith) [inline .npmignore]: what do you think about moving these to a "rollup/" folder? or a "build" or "dist" folder?

File: package.json [hunk 3]
```diff
@@ -20,23 +20,32 @@
...
     "smtp-server": "^3.4.6"
   },
   "engine": [
     "node >= 6"
   ],
   "main": "email.js",
   "scripts": {
-    "test": "mocha -R spec"
+    "test": "mocha",
+    "rollup": "rollup -c rollup.config.js && mocha --file email.bundle.test.js"
```
Comment (eleith) [inline package.json]: could these be separated to a "rollup" and "rollup-test"? or is it preferred to have it happen in one step?

File: package.json [hunk 4]
```diff
@@ -20,23 +20,32 @@
...
+    "prettier": "^1.13.7",
+    "rollup": "^0.62.0",
+    "rollup-plugin-commonjs": "^9.1.3",
+    "rollup-plugin-node-resolve": "^3.3.0",
     "smtp-server": "^3.4.6"
   },
   "engine": [
     "node >= 6"
   ],
   "main": "email.js",
```
Comment (eleith) [inline package.json]: if we move this library over to use the build, does it just require changing "main" ? could you walk me through what we would need to do and what the impact of switching to this is for end users?

File: Readme.md [hunk 5]
```diff
@@ -205,7 +206,7 @@ associative array of currently supported SMTP authentication mechanisms
         headers     // object containing header=>value pairs for inclusion in this attachment's header
         related     // an array of attachments that you want to be related to the parent attachment
     }
-	
+
 ## Authors
 
 eleith
```
Comment (eleith) [inline Readme.md]: add your username here please =]

Comment (zackschuster) [reply Readme.md]: =]

File: package.json (hunk 4 above)
Comment (zackschuster) [reply package.json]: yes, `main` would need to be updated to point at the rollup build. end users shouldn't notice a difference* or need to adjust their own code.

*one relatively minor exception: the bundling apparently puts `console` into the global scope for the `text-encoding` library, which leads to a bunch of unnecessary logging about some piece of upstream code specifying an encoding in the constructor. harmlessly annoying and totally fixable on its own (probably by doing AST parsing/string replacement with the output from rollup's JS API) but wasn't a priority to me for this PR.

File: package.json (hunk 3 above)
Comment (zackschuster) [reply package.json]: done :)

File: .npmignore (hunk 2 above)
Comment (zackschuster) [reply .npmignore]: moved to `rollup/`

File: .travis.yml (hunk 1 above)
Comment (zackschuster) [reply .travis.yml]: yes; travis is auto-detecting the yarn.lock file for install so i kept the streak going. i've updated config to use npm.

**Comment by zackschuster** [comment]
follow-up: i've been running the bundle in prod for the last few weeks & it works perfectly. zero issues to report.

**Comment by eleith** [comment]
thanks again!

**Pull request merged**